pkg/utils/utils.go

```diff
@@ -449,3 +457,164 @@ func DoesMigrationScheduleMigrateNamespaces(migrationSchedule stork_api.Migratio
 	}
 	return found, nil
 }
+
+// get Pod from PVC details
+func GetPodFromPVC(pvcName, namespace string) (*v1.Pod, error) {
+	fn := "GetPodFromPVCName"
+	// create pod list to store the relevant pods
+	var podList []*v1.Pod
+	pods, err := core.Instance().GetPodsUsingPVC(pvcName, namespace)
+	if err != nil {
+		errMsg := fmt.Sprintf("error fetching pods using PVC %s/%s: %v", namespace, pvcName, err)
+		logrus.Errorf("%s: %v", fn, errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+	// filter out the pods that are created by us
+	for _, pod := range pods {
+		labels := pod.ObjectMeta.Labels
+		if _, ok := labels[kdmpDriver.DriverNameLabel]; ok {
+			continue
+		}
+		podList = append(podList, &pod)
+	}
+
+	if len(podList) == 0 {
+		errMsg := fmt.Sprintf("no application pod found for PVC %s/%s", namespace, pvcName)
+		logrus.Debugf("%s: %v", fn, errMsg)
+		return nil, fmt.Errorf(errMsg)
+	} else if len(podList) > 1 {
+		logrus.Warnf("%s: multiple application pods found using common PVC %s/%s", fn, namespace, pvcName)
+		// TODO we need to handle this case of PVC being used by multiple pods in the same namespace
+		// For now we will return the first pod in the list as a placeholder
+	}
+	return podList[0], nil
+}
+
+func getIdFromSecurityContext(securityContext interface{}) (int64, int64) {
+	uid := UndefinedId
+	gid := UndefinedId
+
+	// Baseline or privilege PSA can have nil security Context.
+	if securityContext == nil {
+		return uid, gid
+	}
+	switch s := securityContext.(type) {
+	case *v1.SecurityContext:
+		if s != nil {
+			if s.RunAsUser != nil {
+				uid = int64(*s.RunAsUser)
+			}
+			if s.RunAsGroup != nil {
+				gid = int64(*s.RunAsGroup)
+			}
+		}
+	case *v1.PodSecurityContext:
+		if s != nil {
+			if s.RunAsUser != nil {
+				uid = int64(*s.RunAsUser)
+			}
+			if s.RunAsGroup != nil {
+				gid = int64(*s.RunAsGroup)
+			}
+		}
+	default:
+		// Handle the default case here
+		logrus.Debugf("Unknown type of security context obtained: %v", s)
+	}
+	return uid, gid
+}
+
+// GetPodUserId: Get the UID/GID of the application pod which is using the PVC for which backup is triggered.
+func GetPodUserId(pod *v1.Pod) (int64, int64) {
```

Comment:
One other user we might also need to think of is having runAsUser sepcified at PodSecurity Context level and runAsGroup defined at container security context level which i feel might be used by customers.
In such a case we need to pick needed values from each one of them.

```
spec:
  securityContext:
    runAsUser: 1000  
    fsGroup: 2000    
  containers:
    securityContext:
      runAsGroup: 3000
```

Reply:
Yeah I got it, but these values primarily we get it for Job Pod to run gracefully  So if we get any one i.e. uid or gid we should be good. Else the code block for if else will become too complex. I agree here that we will leave any setting present in PodSecurity Context level if we find anything at container level. Let me take care of it post FC.